spark-atlas-connector/src/main/scala/com/hortonworks/spark/atlas/ml/MLPipelineTracker.scala

```diff
@@ -31,5 +47,196 @@ class MLPipelineTracker(
     this(new AtlasClientConf)
   }
 
+  private val capacity = conf.get(AtlasClientConf.BLOCKING_QUEUE_CAPACITY).toInt
+
+  // A blocking queue for Spark Listener ExternalCatalog related events.
+  @VisibleForTesting
+  private[atlas] val eventQueue = new LinkedBlockingQueue[SparkListenerEvent](capacity)
+
+  private val timeout = conf.get(AtlasClientConf.BLOCKING_QUEUE_PUT_TIMEOUT).toInt
+
+  @VisibleForTesting
+  @volatile private[atlas] var shouldContinue: Boolean = true
+
+  private val cachedObjects = new mutable.HashMap[String, Object]
+
+  private val uri = "hdfs://"
+
+  private var tableEntity1:Seq[AtlasEntity] = null
+
+  startThread()
+
+  override def onOtherEvent(event: SparkListenerEvent): Unit = {
+    if (!shouldContinue) {
+      // No op if our tracker is failed to initialize itself
+      return
+    }
+
+    // We only care about ML related events.
+    event match {
+      case e: MLListenEvent =>
```

Comment:
I think it would be better to use reflection for all the ML related events, so that the code can be compiled with Apache Spark.

Reply:
sure, we can update this later.

Reply:
@jerryshao, we need to merge this at first since the code improvement is putted into later step. @yanboliang @bikassaha

Reply:
Sure. let me propose a revision on this

Reply:
hi @jerryshao  we almost done the reflection on this part.  even if we pass the build process, it still fail to pass the test cases. right?

Reply:
@jerryshao  the reflection is done.